pkg/commatrix-creator/commatrix_test.go

```diff
@@ -0,0 +1,342 @@
+				},
+			},
+			Type: corev1.ServiceTypeNodePort,
+		},
+	}
+
+	testEndpointSlice = &discoveryv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-service-endpoints",
+			Namespace: "test-ns",
+			Labels: map[string]string{
+				"kubernetes.io/service-name": "test-service",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Kind: "Service",
+					Name: "test-service",
+				},
+			},
+		},
+		AddressType: discoveryv1.AddressTypeIPv4,
+		Endpoints: []discoveryv1.Endpoint{
+			{
+				NodeName:  &testNode.Name,
+				Addresses: []string{"192.168.1.1", "192.168.1.2"},
+			},
+		},
+		Ports: []discoveryv1.EndpointPort{
+			{
+				Name:     nil,
+				Port:     func(i int32) *int32 { return &i }(80),
+				Protocol: func(p corev1.Protocol) *corev1.Protocol { return &p }(corev1.ProtocolTCP),
+			},
+		},
+	}
+)
+
+var _ = g.Describe("Commatrix creator pkg tests", func() {
+	g.Context("Get ComDetails List From File", func() {
+		for _, format := range []string{types.FormatCSV, types.FormatJSON, types.FormatYAML} {
+			g.It(fmt.Sprintf("Should successfully extract ComDetails from a %s file", format), func() {
+				g.By(fmt.Sprintf("Creating new communication matrix with %s static entries format", format))
+				cm, err := New(nil, fmt.Sprintf("../../samples/custom-entries/example-custom-entries.%s", format), format, 0, 0)
+				o.Expect(err).ToNot(o.HaveOccurred())
+
+				g.By("Getting ComDetails List From File")
+				gotComDetails, err := cm.GetComDetailsListFromFile()
+				o.Expect(err).ToNot(o.HaveOccurred())
+
+				g.By("Comparing gotten ComDetails to wanted ComDetials")
+				o.Expect(gotComDetails).To(o.Equal(exampleComDetailsList))
+			})
+		}
+
+		g.It("Should return an error due to non-matched customEntriesPath and customEntriesFormat types", func() {
+			g.By("Creating new communication matrix with non-matched customEntriesPath and customEntriesFormat")
+			cm, err := New(nil, "../../samples/custom-entries/example-custom-entries.csv", types.FormatJSON, 0, 0)
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Getting ComDetails List From File")
+			gotComDetails, err := cm.GetComDetailsListFromFile()
+			o.Expect(err).To(o.HaveOccurred())
+
+			g.By("Comparing gotten ComDetails to empty ComDetials")
+			o.Expect(gotComDetails).To(o.Equal(nilComDetailsList))
+		})
+
+		g.It("Should return an error due to an invalid customEntriesFormat", func() {
+			g.By("Creating new communication matrix with invalid customEntriesFormat")
+			cm, err := New(nil, "../../samples/custom-entries/example-custom-entries.csv", types.FormatNFT, 0, 0)
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Getting ComDetails List From File")
+			gotComDetails, err := cm.GetComDetailsListFromFile()
+			o.Expect(err).To(o.HaveOccurred())
+
+			g.By("Comparing gotten ComDetails to empty ComDetials")
+			o.Expect(gotComDetails).To(o.Equal(nilComDetailsList))
+		})
+	})
+
+	g.Context("Get static enteries from file", func() {
```

Comment:
Typo in entries

Reply:
This is static entries only matrix?

Reply:
Yes, because we only test getting the static entries here.